lint/unnamedResult_checker.go

```diff
@@ -0,0 +1,77 @@
+package lint
+
+import (
+	"go/ast"
+	"go/types"
+)
+
+func init() {
+	addChecker(unnamedResultChecker{}, &ruleInfo{})
+}
+
+type unnamedResultChecker struct {
+	baseFuncDeclChecker
+}
+
+func (c unnamedResultChecker) New(ctx *context) func(*ast.File) {
+	return wrapFuncDeclChecker(&unnamedResultChecker{
+		baseFuncDeclChecker: baseFuncDeclChecker{ctx: ctx},
+	})
+}
+
+func (c *unnamedResultChecker) CheckFuncDecl(decl *ast.FuncDecl) {
+	results := decl.Type.Results
+	switch {
+	case results == nil || c.resultsNum(results.List) < 2:
+		return
+
+	case c.resultsNum(results.List) == 2:
+		if !c.isBoolOrError(results.List[0].Type) &&
+			c.isBoolOrError(results.List[1].Type) {
+			// no need to name results for (T, error) or (T, bool)
+		} else {
+			c.warn(decl)
+		}
+
+	default:
+		for _, res := range results.List {
+			if len(res.Names) == 0 {
+				c.warn(decl)
+				break
+			}
+		}
+	}
+}
+
+func (c *unnamedResultChecker) warn(n ast.Node) {
+	c.ctx.Warn(n, "consider to give name to results")
+}
+
+// resultsNum will return number of results.
+// If they're unnamed than we meed to return number of fields.
+// Else we need to sum all amount of names in each field.
+func (c *unnamedResultChecker) resultsNum(fields []*ast.Field) int {
```

Comment:
Almost looks like `FieldList.NumFields`.
Can we use that instead or I'm missing something?